rsocket-core/src/main/java/io/rsocket/RSocketFactory.java

```diff
@@ -450,6 +466,11 @@ public ServerRSocketFactory errorConsumer(Consumer<Throwable> errorConsumer) {
       return this;
     }
 
+    public ServerRSocketFactory lease(BiConsumer<Leases, LeaseOptions> leaseSenderConsumer) {
+      this.leaseSenderConsumer = leaseSenderConsumer;
```

Comment:
@robertroeser LeaseOptions accept callback for Lease use events (accept/reject), and is separated from lease sender/receiver